department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-17509: Mocking BGS Contentions in Test Data

Resolved [APPEALS-17509](https://vajira.max.gov/browse/APPEALS-17509)

### Description
Added MST and PACT information to Contention and BGS Contention generators, as well as adjusted BGSServiceRecordMaker to create MST and PACT Higher Level Reviews/Request Issues with Contentions. 

### Acceptance Criteria
- [ ] Contention Generator creates contentions with MST
- [ ] Contention Generator creates contentions with PACT
- [ ] BGS Contention Generator creates contentions with MST
- [ ] BGS Contention Generator creates contentions with PACT
- [ ] BGSServiceRecordMaker generates End Product Establishments that have Request Issues with Contentions (BGS and MST)
- [ ] EPEs generated show up at **Contestable Issues** when intaking an appeal.

### Testing Plan
1. Go to the Rails console by running `cd ~/appeals/caseflow` followed by `rails console`.
#### Testing Contention Generator
2. To create an MST contention, run `contention = Generators::Contention.build_mst_contention`.
3. Confirm you can access the MST status by running `contention.special_issues[:code]`
![image](https://user-images.githubusercontent.com/99915461/232489872-27302837-24e5-4913-90e6-f47e60614cb7.png)
4. Repeat steps 2 and 3 for PACT by running `contention = Generators::Contention.build_pact_contention` and accessing the PACT status by running `contention.special_issues[:code]`
![image](https://user-images.githubusercontent.com/99915461/232490327-ed7ebb3e-3edf-4748-822f-71528ae643a6.png)

#### Testing BGS Contention Generator
1. To create an MST BGS contention, run `contention = Generators::BgsContention.build_mst_contention`.
2. Confirm you can access the MST status by running `contention.special_issues[:spis_tc]`
![image](https://user-images.githubusercontent.com/99915461/232491047-c720a4ae-b9d3-474e-b046-0dd5e07ff586.png)
3. Repeat steps 2 and 3 for PACT by running `contention = Generators::BgsContention.build_pact_contention` and accessing the PACT status by running `contention.special_issues[:spis_tc]`
![image](https://user-images.githubusercontent.com/99915461/232491258-6eeca134-9886-49ff-83bf-08eca470f233.png)


#### Testing BgsServiceRecordMaker
1. In the Rails Console, run `Fakes::BGSServiceRecordMaker.new.call` 
2. When the script is finished running, you should see new Higher Level Reviews created in the database, one with MST and one with PACT contentions. Use any of these veteran file numbers to confirm.
992190636
107924012
329146737
981775045
738464961
245620198
955197769
473949326
933747849
696650437
598042239
787148925
657332982
416322111
968514949
779309925
169397130
276880282
138034137
192920593
190595135
3. To confirm that the End Product Establishment has contention in the Rails console, run `epe = EndProductEstablishment.find_by(veteran_file_number: 992190636)` followed by `hlr.contentions`
![image](https://user-images.githubusercontent.com/99915461/232493404-930be97d-c4ba-498f-8a54-e86e9344af2b.png)
4. To confirm request issues have a contention linked to them, run `epe.request_issues[0].contention` to view the contention.
![image](https://user-images.githubusercontent.com/99915461/232494085-fbe6f9ff-1733-49e1-a587-493a68168266.png)

#### Confirming in the Intake UI
1. Log in as an intake user (I used `BVADWISE`) and navigate to **Mail Intake**
2. Select the first option for Decision Review from the dropdown and click **continue to search**
![image](https://user-images.githubusercontent.com/99915461/232495643-9b421c74-c6f6-4c53-84f0-9f45476108ec.png)
3. In the search bar, enter a veteran file number from the list above and click the magnifying glass.
![image](https://user-images.githubusercontent.com/99915461/232495910-57112271-9cd4-4a9d-98d5-fbb3479d3e6f.png)
4. Fill out the form and click **continue to next step**.
![image](https://user-images.githubusercontent.com/99915461/232496025-de51e855-f254-4e9c-8b7b-4dba4d049ead.png)
5. Click the **Add issues** button. You should see a list of past decisions populating from the generator. 
![image](https://user-images.githubusercontent.com/99915461/232496291-4dd341e7-ee81-418d-9ae1-02ce27383dea.png)
